### DISCUSSION TITLE
Chlorinator Pool/Spa output listed backwards?

### DISCUSSION THREAD

**Discussion by plewenson** [tagyoureit/nodejs-poolController#511] [App and O/S Config Questions]
Hi!  Fantastic app and I'm really excited to finally have control over my pool this summer!  

Everything seems to be working extremely well, except I'm seeing the target output values of my cholrinator's pool vs. spa are inconsistent.  At my EasyTouch 8 panel, it's listed correctly as Pool = 45% and Spa = 2%.  

From the attached images, DashPanel's chemistry settings screen show these values correctly.

However, DashPanel's main screen lists them backwards as does MQTT Explorer at Spa = 45% and Pool = 2% (which impacts HA metrics).

Any thoughts?

Controller: EasyTouch 8
Chlorinator:  Intellichlor IC40
NJPC, DashPanel, MQTT and HomeAssistant via Docker on a Synology Server
RS485 Adapter to Wifi

![MQTT Explorer Intellichlor](https://user-images.githubusercontent.com/42219601/176467485-136b7efa-1beb-4589-96dd-3ae4512b0765.png)
![Dashpanel Chemistry Settings Intellichlor](https://user-images.githubusercontent.com/42219601/176467488-24c2f2ad-3b7d-40e2-904b-17df7a4acb3d.png)
![Dashpanel Main Screen Intellichlor](https://user-images.githubusercontent.com/42219601/176467489-68f03e3b-7d0c-4c5c-8a09-594b4783590d.png)

**Comment by tagyoureit** [comment]
Interesting.  We have a lot of users with Intellichlor and have not seen the values reversed.  Can you do a [packet capture](https://github.com/tagyoureit/nodejs-poolController/wiki/How-to-capture-all-packets-for-issue-resolution#-dashpanel) and upload it here?

Let the capture run for 1 minute with your pool filter.  Then change the chlor pool setpoint to something else and let it run for another minute.  Then change to the spa and repeat (run for 1 min; then change spa setpoint and run for 1 more min).

**Reply by plewenson** [reply]
Thanks so much for the quick reply!  I've attached 2 versions of the packet capture as you requested.  

The first version I started the capture and then changed the pool/spa values using DashPanel.

The second capture I changed the IC40's values directly from my EasyTouch system.

What I watched via MQTT Explorer is when I changed values via Dashpanel, the values tracked correctly.

When I changed via EasyTouch, MQTT has them backwards and Dashpanel main screen shows them backwards too (with the Chemistry section being correctly shown as per EasyTouch).

Hope this helps!

[replay from EasyTouch values.zip](https://github.com/tagyoureit/nodejs-poolController/files/9013078/replay.from.EasyTouch.values.zip)
[replay from Dashpanel values.zip](https://github.com/tagyoureit/nodejs-poolController/files/9013079/replay.from.Dashpanel.values.zip)

**Reply by tagyoureit** [reply]
Something is majorly jacked up here.  I didn't even pick up on it earlier but your spa is completely missing from dashPanel in your initial post.  And now looking at your replays the spa and pool are both on at the same time.  This is an impossible condition.  You can never have both bodies of water active at the same time on a shared body.

![2022-06-29_12-53-31](https://user-images.githubusercontent.com/1646840/176533365-393d681c-ef7f-47f6-93ff-b7ff2e6525cd.png)
 
Before I dive into this any further, please do the following.
1. Shut down njsPC.
2. Go out to your EasyTouch and either power cycle it with the circuit breaker or hit the reset button.
3. On your njsPC machine, delete/rename the poolConfig.json/poolState.json files in the `./data` directory.  (You can always get them back from the replays you saved.)
4. Start up njsPC.
5. See if the same conditions happen or if a reset fixed it.

If you still see the issues, upload a new replay and I'll dig into it further.
